ui/src/main/java/org/jtaccuino/core/ui/api/ToggleAction.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2025 JTaccuino Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jtaccuino.core.ui.api;
+
+import javafx.beans.property.ReadOnlyBooleanProperty;
+import javafx.event.ActionEvent;
+import javafx.scene.input.KeyCombination;
+
+public final class ToggleAction implements Action {
+
+    private Action activeAction = null;
```

Comment:
initial assignment is not needed strictly speaking as it is set by the constructor